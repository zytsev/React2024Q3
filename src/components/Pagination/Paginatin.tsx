import style from './pagination.module.css';
import { PropsPagination } from '../../assets/types/types';
import { useContext } from 'react';
import { Context } from '../Context/Context';

export const Pagination = (props: PropsPagination) => {
  const context = useContext(Context);
  const { arrFromApi, setCardsOnPagina, cardsOnPagina } = props;
  const activePagina = context?.activePagina || 1;

  let numberOfPaginas;
  arrFromApi
    ? (numberOfPaginas = Math.ceil(arrFromApi.length / cardsOnPagina))
    : 1;
  const arrPaginas = [...Array(numberOfPaginas).keys()].map((i) => i + 1);

  return (
    <div className={style.PGNroot}>
      <svg
        onClick={
          activePagina !== 1
            ? () => context?.setActivePagina(activePagina - 1)
            : () => context?.setActivePagina(activePagina)
        }
        xmlns="http://www.w3.org/2000/svg"
        className={style.PGNsvg}
        viewBox="0 0 512 512"
      >
        <path
          fill="none"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="48"
          d="M244 400L100 256l144-144M120 256h292"
        />
      </svg>
      <ul className={style.PGNul}>
        {arrPaginas.map((item) => (
          <li
            key={item}
            onClick={() => context?.setActivePagina(item)}
            className={activePagina === item ? style.PGNliActive : style.PGNli}
          >
            {item}
          </li>
        ))}
      </ul>
      <svg
        onClick={
          activePagina !== arrPaginas.length
            ? () => context?.setActivePagina(activePagina + 1)
            : () => context?.setActivePagina(activePagina)
        }
        xmlns="http://www.w3.org/2000/svg"
        className={style.PGNsvg}
        viewBox="0 0 512 512"
      >
        <path
          fill="none"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="48"
          d="M268 112l144 144-144 144M392 256H100"
        />
      </svg>
      <label>
        {' '}
        Show on page:
        <select
          value={props.cardsOnPagina}
          onChange={(e) => setCardsOnPagina(Number(e.target.value))}
        >
          <option>3</option>
          <option>6</option>
          <option>9</option>
        </select>
      </label>
    </div>
  );
};
