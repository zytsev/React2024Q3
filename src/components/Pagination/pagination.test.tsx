import { render } from '@testing-library/react';
import { Pagination } from './Paginatin';

import { vitest } from 'vitest';
import { BrowserRouter } from 'react-router-dom';

const exampleCard = {
  id: 7,
  raiting: 7,
  category: 'Coffee',
  imgClass: 'list-img-wrapper3',
  title: 'Honey raf',
  text: 'Espresso with frothed milk, cream and aromatic honey',
  price: 5.5,
};

vitest.mock('../../services/redux/store/store');

describe('Pagination component', () => {
  it('should render element with  passed 1  card ', () => {
    const component = render(
      <BrowserRouter>
        <Pagination arrFromApi={[exampleCard]} />
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
