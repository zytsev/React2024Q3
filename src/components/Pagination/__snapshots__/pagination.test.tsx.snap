// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination component > should render element with  passed 1  card  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_PGNroot_1875ff"
      >
        <svg
          class="_PGNsvg_1875ff"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M244 400L100 256l144-144M120 256h292"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="48"
          />
        </svg>
        <ul
          class="_PGNul_1875ff"
        >
          <li
            class="_PGNliActive_1875ff"
            data-testid="1"
          >
            1
          </li>
        </ul>
        <svg
          class="_PGNsvg_1875ff"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M268 112l144 144-144 144M392 256H100"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="48"
          />
        </svg>
        <label>
           
          Show on page:
          <select>
            <option>
              3
            </option>
            <option>
              6
            </option>
            <option>
              9
            </option>
          </select>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_PGNroot_1875ff"
    >
      <svg
        class="_PGNsvg_1875ff"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M244 400L100 256l144-144M120 256h292"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="48"
        />
      </svg>
      <ul
        class="_PGNul_1875ff"
      >
        <li
          class="_PGNliActive_1875ff"
          data-testid="1"
        >
          1
        </li>
      </ul>
      <svg
        class="_PGNsvg_1875ff"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M268 112l144 144-144 144M392 256H100"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="48"
        />
      </svg>
      <label>
         
        Show on page:
        <select>
          <option>
            3
          </option>
          <option>
            6
          </option>
          <option>
            9
          </option>
        </select>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PopUp component > should render element with  passed 1  card  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_PGNroot_1875ff"
      >
        <svg
          class="_PGNsvg_1875ff"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M244 400L100 256l144-144M120 256h292"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="48"
          />
        </svg>
        <ul
          class="_PGNul_1875ff"
        >
          <li
            class="_PGNliActive_1875ff"
            data-testid="1"
          >
            1
          </li>
        </ul>
        <svg
          class="_PGNsvg_1875ff"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M268 112l144 144-144 144M392 256H100"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="48"
          />
        </svg>
        <label>
           
          Show on page:
          <select>
            <option>
              3
            </option>
            <option>
              6
            </option>
            <option>
              9
            </option>
          </select>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_PGNroot_1875ff"
    >
      <svg
        class="_PGNsvg_1875ff"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M244 400L100 256l144-144M120 256h292"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="48"
        />
      </svg>
      <ul
        class="_PGNul_1875ff"
      >
        <li
          class="_PGNliActive_1875ff"
          data-testid="1"
        >
          1
        </li>
      </ul>
      <svg
        class="_PGNsvg_1875ff"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M268 112l144 144-144 144M392 256H100"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="48"
        />
      </svg>
      <label>
         
        Show on page:
        <select>
          <option>
            3
          </option>
          <option>
            6
          </option>
          <option>
            9
          </option>
        </select>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
